packages/botonic-plugin-contentful/src/contentful/manage/manage-entry.ts

```diff
@@ -0,0 +1,176 @@
+    readonly manage: ClientAPI,
+    readonly environment: Promise<Environment>
+  ) {}
+
+  async updateFields(
+    context: ManageContext,
+    contentId: ContentId,
+    fields: { [contentFieldType: string]: any }
+  ): Promise<{ [contentFieldType: string]: any }> {
+    const environment = await this.environment
+    const oldEntry = await this.getEntry(environment, contentId)
+    let needUpdate = false
+    for (const key of Object.keys(fields)) {
+      if (!isOfType(key, ContentFieldType)) {
+        throw new CmsException(`'${key}' is not a valid content field type`)
+      }
+      const field = this.checkOverwrite(context, oldEntry, key, false)
+      if (oldEntry.fields[field.cmsName][context.locale] === fields[key]) {
+        continue
+      }
+      needUpdate = true
+      oldEntry.fields[field.cmsName][context.locale] = fields[key]
+    }
+    if (!needUpdate) {
+      return oldEntry.fields
+    }
+    // we could use this.deliver.contentFromEntry & IgnoreFallbackDecorator to convert
+    // the multilocale fields returned by update()
+    await this.writeEntry(context, oldEntry)
+    return oldEntry.fields
+  }
+
+  async createContent(
+    context: ManageContext,
+    model: ContentType,
+    id: string
+  ): Promise<void> {
+    const environment = await this.environment
+    try {
+      await environment.createEntryWithId(model, id, {
+        fields: {},
+      })
+    } catch (e) {
+      throw new CmsException('ERROR while creating content', e)
+    }
+  }
+
+  async deleteContent(
+    context: ManageContext,
+    contentId: ContentId
+  ): Promise<void> {
+    const environment = await this.environment
+    try {
+      const oldEntry = await this.getEntry(environment, contentId)
+      if (oldEntry.isPublished()) await oldEntry.unpublish()
+      await oldEntry.delete()
+    } catch (e) {
+      throw new CmsException('ERROR while deleting content', e)
+    }
+  }
+
+  private checkOverwrite(
+    context: ManageContext,
+    entry: Entry,
+    fieldType: ContentFieldType,
+    failIfMissing: boolean
+  ): ContentField {
+    if (entry.isArchived()) {
+      throw new CmsException(`Cannot update an archived entry`)
+    }
+    const field = CONTENT_FIELDS.get(fieldType)
+    if (!field) {
+      throw new CmsException(`Invalid field type ${fieldType}`)
+    }
+    if (!context.locale) {
+      // paranoic check
+      throw new Error('Context.locale must be defined')
+    }
+    if (!(field.cmsName in entry.fields)) {
+      if (!failIfMissing) {
+        entry.fields[field.cmsName] = {}
+        return field
+      }
+      const fields = Object.keys(entry.fields)
+      throw new CmsException(
+        `Field '${field.cmsName}' not found in entry of type '${
+          entry.sys.contentType.sys.id
+        }. It only has ${JSON.stringify(fields)}'`
+      )
+    }
+    if (!context.allowOverwrites) {
+      const value = entry.fields[field.cmsName][context.locale]
+      if (value) {
```

Comment:
simplify it to make it readable :)
```
const error = `Cannot overwrite field '${field.cmsName}' of entry '${entry.sys.id}'
const detail = `"(has value '${String(value)}') because ManageContext.allowOverwrites is false``
throw new CmsException(error + detail)
```